### ISSUE TITLE
Clean up `SwiftCompile` task properties to use `Property`

### ISSUE THREAD

**Issue by big-guy** [gradle/gradle-native#308]
We should use `Property` in all cases (context #295)

**Comment by big-guy** [comment]
linking and other Swift tasks

**Comment by big-guy** [comment]
Here's a PR:
https://github.com/gradle/gradle/pull/4319

It converts all of the properties of `SwiftCompile`, `AbstractLinkTask` (and subclasses), `InstallExecutable`, `ExtractSymbols`, `StripSymbols`, `CreateStaticLibrary` to use the Provider API.  To keep the existing integration tests working, I also converted `Assemble` and `AbstractNativeCompileTask` to use the Provider API for targetPlatform and toolChain.  This is a breaking change in a number of ways (all to `@Incubating` types):
- Return types changed from `T` to `Property<T>`
- Setters were removed 
- Some properties were renamed to use a consistent name everywhere (e.g., `platform` -> `targetPlatform`) or to eliminate confusion (what's a `binaryFile` for a link task? what's a `sourceFile` for an install task?)

To complete #295, we would need to pass through all of the native tasks (from the software model) and convert those as well.
